content/interfaces/reading/interfaces/index.md

```diff
@@ -0,0 +1,219 @@
+
+```java{linenos=table,hl_lines=[],linenostart=1}
+public class PetOwner
+{
+    private Feedable pet;
+
+    public PetOwner(Feedable pet) {
+        this.pet = pet;
+    }
+
+    public void feedThePet() {
+
+        // ...code to prepare the pet's meal...
+
+        pet.eat();
+    }
+}
+
+public class CatOwner extends PetOwner
+{
+    // code that requires Cat-specific behavior
+}
+```
+
+We’ve created a `PetOwner` class that encapsulates the behavior for any pet (any `Feedable`, actually), and have `CatOwner` extend `PetOwner`. This allows other classes to extend `PetOwner` to make, say, a `DogOwner` that knows how to play fetch with their pet, or a `HorseOwner` that knows how to ride their pet. It also reduces the dependency of the `feedThePet` method on the specific type of pet, since it doesn’t need to care.
+
+To use this new class design, we can revise the sample code from above as follows:
+
+```java{linenos=table,hl_lines=[],linenostart=1}
+HouseCat suki = new HouseCat("Suki", 12);
+CatOwner Annie = new CatOwner(suki);
+
+Annie.feedThePet();
+```
+
+While the code usage here remains unchanged except for changing the method name from `feedTheCat` to the more generic `feedThePet`, the opportunities for using the classes we’ve built are much wider since the defined classes are no longer dependent on the specific `Cat` class. Also notice that we’ve used the object `suki` in a polymorphic way, creating it as a `HouseCat`, but using it as a `Feedable` within the `CatOwner` class.
+
+{{% notice blue "Note" "rocket" %}} 
+ Like inheritance, interfaces enable polymorphic usage of objects. We can create an object and then use it in different contexts based on the interfaces that it implements.
+{{% /notice %}}
+
+_Interfaces may not be created like objects are, with_ `new`. You may implement an interface, or declare variables and parameters as interface types. You can not, however, create an instance of an interface.
+
+## Benefits of Using Interfaces
+
+Once you get used to interfaces, you’ll begin to think more abstractly about which _behaviors_ your code requires rather than which _classes_ your code requires. This means you’ll be able to “code to interfaces” (an OOP principle) instead of coding to classes, and your code will become more flexible and extensible.
+
+Here are a few benefits of using interfaces:
+
+   1. You can only extend one class, but you may implement many interfaces.
+
+   1. You can extend a class and implement an interface at the same time.
+
+   1. By declaring variables and parameters as interface types, you make your code useful for a much wider variety of situations.
+
+   1. When you declare properties and return types to be interface types, you decouple code using your classes from the actual class types you use. This means that you are free to change the specific implementation of your classes without affecting those using them.
+
```

Comment:
Suggestion - remove extra spaces to match formatting of other ordered lists on page.